### ISSUE TITLE
isodow date part in EXTRACT() not supported

### ISSUE THREAD

**Issue by lukaseder** [cockroachdb/cockroach#42936]
**Describe the problem**

This may well be a feature request, but since the documentation mentions this should work, I have reported a bug.

In PostgreSQL, I can extract the ISO day of week as follows:

```sql
select extract(isodow from current_timestamp)
```

According to the documentation, this should be possible in CockroachDB as well:
https://www.cockroachlabs.com/docs/stable/functions-and-operators.html#date-and-time-functions

![image](https://user-images.githubusercontent.com/734593/70139098-0d284280-1692-11ea-9709-9c54a7d903fb.png)

However, I'm getting:

```
SQL Error [22023]: ERROR: extract(): unsupported timespan: isodow
```

A few others are also not supported, including:

- Millenium (easy to emulate)
- Century (easy to emulate)
- Decade (easy to emulate)
- Isoyear
- Julian

**Environment:**
 - CockroachDB version: CockroachDB CCL v19.2.1 (x86_64-unknown-linux-gnu, built 2019/11/18 23:23:55, go1.12.12)
 - Server OS: Linux on Docker on Windows
 - Client app: JDBC

**Comment by otan** [comment]
it appears this is a typo in the docs...

it is supported as of https://github.com/cockroachdb/cockroach/pull/41784/files (and is on master!), but currently not planning on backporting this to 19.2.2. is it necessary to? not sure about triggering documentation updates.

**Comment by awoods187** [comment]
@otan if this is low risk it'd be nice to backport to 19.2.2 so that jOOQ could take advantage of it. cc @jordanlewis to weigh in on the risk.

As far as docs go, you can file a bug here (https://github.com/cockroachdb/docs/issues). I'd wait till we decide if we are backporting or not before filing.

**Comment by otan** [comment]
this one is backportable quite safely imo!

**Comment by otan** [comment]
I lied - there's ONE danger to backporting that PR in it's entirety (https://github.com/cockroachdb/cockroach/pull/41784).

It changes a fundamental behaviour people may expect from `extract` in 19.2. in particular, `milliseconds`, and `microseconds` now include the `seconds` component (e.g. second 59.234 for extract(`millisecond`) used to return `234`, but now returns `59234` to match postgres). Of course, i can isolate out that particular change in the backport. What do you think?

**Comment by otan** [comment]
ah wait, i think that may just be rebasing badly. let me dig further.

**Comment by otan** [comment]
Judging by the backport, this is a docs bug. The documentation for `isodow` is only added by the backporting PR #42957 to 19.2.

Does `stable` point at master? https://www.cockroachlabs.com/docs/v19.1/functions-and-operators.html#date-and-time-functions (note the v19.1) does not document this at all.

**Comment by otan** [comment]
With cockroachdb/docs#6108 closed to fix the docs, and backport https://github.com/cockroachdb/cockroach/pull/42957 merged for 19.2.3 (probably won't make 19.2.2), i'm closing this.
